code/modules/integrated_electronics/subtypes/manipulation.dm

```diff
@@ -408,9 +417,9 @@
 	)
 	spawn_flags = IC_SPAWN_RESEARCH
 	power_draw_per_use = 50
-	var/max_w_class = WEIGHT_CLASS_NORMAL
 
 /obj/item/integrated_circuit/manipulation/thrower/do_work()
+	var/max_w_class = assembly.w_class
```

Comment:
change line if(max_w_class && (A.w_class > max_w_class))
to if(max_w_class && (A.w_class >= max_w_class))

Reply:
it's intended for the thrower to be able to throw items the size of the assembly, though? as in, bulky assemblies throw bulky things, normal assemblies throw normal stuff, small assemblies throw small stuff

Reply:
well okay.

Reply:
Honestly, at first it was a bug.But it was so hilarious, so I decided to leave it.

Reply:
Remember, the creeper was also a bug and turned into the face of the entire game it was made for!